### PR TITLE
added Controller layer, encapsulated access to Entity through Dao cla…

### PR DESCRIPTION
**Pull request by IhorKalaur**
…sses, added method getById()

### CODE REVIEW

**Review by irynamekh — CHANGES_REQUESTED** [review]
I can't write comments for some reason, so I'll write here)
In the condition for the task it is said that we must implement Mapstruct, I see you have created only the mapper class, maybe you need to finish it.
It also seems to me that you do not need constructors in dto, just put the Data annotation, lombok will create the constructor automatically

**Comment by IhorKalaur** [comment]
> I can't write comments for some reason, so I'll write here) In the condition for the task it is said that we must implement Mapstruct, I see you have created only the mapper class, maybe you need to finish it. It also seems to me that you do not need constructors in dto, just put the Data annotation, lombok will create the constructor automatically
Implemented Mapstruct. In my previous version, I needed a constructor with parameters, for simplified dto initialization, the Lombok library by default generates a constructor without parameters

**Comment by IhorKalaur** [comment]
> I can't write comments for some reason, so I'll write here) In the condition for the task it is said that we must implement Mapstruct, I see you have created only the mapper class, maybe you need to finish it. It also seems to me that you do not need constructors in dto, just put the Data annotation, lombok will create the constructor automatically

Implemented Mapstruct. In my previous version, I needed a constructor with parameters, for simplified dto initialization, the Lombok library by default generates a constructor without parameters.
Lombok can generate constructors with default parameters using the @AllArgsConstructor annotation

**Review by irynamekh — COMMENTED** [review]
Thanks for the tip, I didn't know about this annotation) Great job!

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/controller/BookController.java]:
```suggestion
@RequestMapping(value = "/api/books")
```

File: src/main/java/com/example/bookstore/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.example.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
+
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/dto/BookDto.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.example.bookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/dto/BookDto.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/dto/BookDto.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/model/Book.java [hunk 5]
```diff
@@ -12,19 +12,30 @@
...
     private String author;
-    @Column(nullable = false,
+    @Column(name = "isbn",
+            nullable = false,
             unique = true)
     private String isbn;
-    @Column(nullable = false)
+    @Column(name = "price",
+            nullable = false,
+            columnDefinition = "BIGINT CHECK (price >= 0)"
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/model/Book.java]: for me it seems redundant

File: src/main/java/com/example/bookstore/model/Book.java [hunk 6]
```diff
@@ -12,19 +12,30 @@
...
+    )
     private BigDecimal price;
+    @Column(name = "description")
     private String description;
+    @Column(name = "coverImage")
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/model/Book.java]:
```suggestion
    @Column(name = "cover_image")
```

File: src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -54,4 +51,15 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Book getById(Long id) {
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/repository/impl/BookRepositoryImpl.java]: return optional instead of throwing an exception

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve few things.

File: src/main/java/com/example/bookstore/model/Book.java (hunk 5 above)
Comment (IhorKalaur) [reply src/main/java/com/example/bookstore/model/Book.java]: I wrote this code before learning about input validation, now I know a better way to do it

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request closed without merging**